linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -3836,6 +3837,7 @@ public void remove() {
     // extra predicate in case of non equi-join, in case of equi-join it will be null
     private final Predicate2<TSource, TInner> extraPredicate;
     private final Function2<TSource, TInner, TResult> resultSelector;
+    private final Comparator<TKey> comparator; // possibly null (compareTo to be used in that case)
     private boolean done;
```

Comment:
Should we put the comment on the top?

Reply:
I have added a javadoc `@param` comment for comparator in mergeJoin method

Reply:
AFAIK, according to the calcite code convention, the comment is always on the top.

Reply:
sorry, I misunderstood you, I will change it

Reply:
changed.